module/VuFind/src/VuFind/Role/PermissionProvider/SessionKey.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+/**
+ * Session key permission provider.
+ *
+ * PHP version 8
+ *
+ * Copyright (C) Catalyst IT 2024.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Authorization
+ * @author   Alex Buckley <alexbuckley@catalyst.net.nz>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     http://www.vufind.org  Main Page
+ */
+
+namespace VuFind\Role\PermissionProvider;
+
+use Laminas\Session\Container;
+
+/**
+ * Session key permission provider.
+ *
+ * @category VuFind
+ * @package  Authorization
+ * @author   Alex Buckley <alexbuckley@catalyst.net.nz>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ */
+
+class SessionKey implements PermissionProviderInterface
+{
+    use \VuFind\Log\LoggerAwareTrait;
+
+    /**
+     * Session container
+     *
+     * @var Container
+     */
+    protected $session;
+
+    /**
+     * Constructor
+     *
+     * @param $session Session container
+     */
+    public function __construct(
+        Container $session
```

Comment:
You can make this more concise by saying `protected Container $session` here, deleting the `protected $session;` definition up above, and removing the `$this->session = $session;` line below. This is called "constructor property promotion" -- it's a relatively new PHP feature designed to make constructors and related logic less verbose, and I'm becoming a big fan of it. :-)

Reply:
Wonderful tip, thanks @demiankatz . Very helpful to know about this great new PHP feature. I've pushed a follow-up making these changes!